### PR TITLE
update permissions to the new schema based on teams and/or level

### PR DESCRIPTION
**Pull request by willemarcel**
- update permissions on the frontend to the new schema based on teams and/or level
- add teams to project summary endpoint
- rename `projectTeams` field to `teams` on the projectsDTO
- update project edit and project code views to fit the new field name

:stop_sign:  To be merged and to work, we depend on some changes on the backend.

### CODE REVIEW

**Comment by ramyaragupathy** [comment]
Completed:
* [x] Migration
* [x] status file update
* [x] field validation check
* [x] DTO udpates
* [x] Model file update

Pending items:
* [ ] Return string values in DTOs
* [ ] Unit test case updates - this is resulting in circle errors
* [ ] Update main functions that decide if a user is permitted to map/validate

cc @willemarcel @xamanu

File: server/models/postgis/statuses.py [hunk 1]
```diff
@@ -127,3 +127,19 @@ class TeamMemberFunctions(Enum):
     EDITOR = 0
     MANAGER = 1
     MEMBER = 2
+
+
+class MappingPermission(Enum):
+    """ Describes a set of permissions for mapping on a project """
+
+    ANY = 0
+    LEVEL = 1
```
Comment (willemarcel) [inline server/models/postgis/statuses.py]: @ramyaragupathy 
```
    TEAMS = 2
    TEAMS_LEVEL = 3
```
should be here too

File: server/services/project_service.py [hunk 2]
```diff
@@ -246,9 +248,56 @@ def is_user_permitted_to_validate(project_id, user_id):
...
+            validation_permission = project.validation_permission
+            if validation_permission == ValidationPermission.TEAMS.value:
+                teams_dto = TeamService.get_project_teams_as_dto(project_id)
+
+                for team_dto in teams_dto.teams:
+                    team_id = team_dto.team_id
+                    team_role = team_dto.role
+                    if (
+                        TeamService.is_user_member_of_team(team_id, user_id)
+                        and team_role == TeamRoles.VALIDATOR.value
```
Comment (willemarcel) [inline server/services/project_service.py]: let's allow PROJECT_MANAGER team roles here too

File: server/services/project_service.py [hunk 3]
```diff
@@ -246,9 +248,56 @@ def is_user_permitted_to_validate(project_id, user_id):
...
+
+            if validation_permission == ValidationPermission.TEAMS_LEVEL.value:
+                teams_dto = TeamService.get_project_teams_as_dto(project_id)
+
+                for team_dto in teams_dto.teams:
+                    team_id = team_dto.team_id
+                    team_role = team_dto.role
+                    if not (
+                        TeamService.is_user_member_of_team(team_id, user_id)
+                        and team_role == TeamRoles.VALIDATOR.value
```
Comment (willemarcel) [inline server/services/project_service.py]: PROJECT_MANAGER here too

File: server/services/project_service.py [hunk 4]
```diff
@@ -185,12 +189,11 @@ def is_user_permitted_to_map(project_id: int, user_id: int):
             return False, MappingNotAllowed.PROJECT_NOT_PUBLISHED
 
         tasks = Task.get_locked_tasks_for_user(user_id)
-        print(tasks)
 
         if len(tasks.locked_tasks) > 0:
             return False, MappingNotAllowed.USER_ALREADY_HAS_TASK_LOCKED
 
-        if project.restrict_mapping_level_to_project:
+        if project.mapping_permission:
```
Comment (willemarcel) [inline server/services/project_service.py]: it's missing to update the mapping permissions, right?

File: server/services/project_service.py [hunk 5]
```diff
@@ -246,9 +248,56 @@ def is_user_permitted_to_validate(project_id, user_id):
...
+                            MappingLevel.INTERMEDIATE.value,
+                            MappingLevel.ADVANCED.value,
+                        ):
+                            return False, ValidatingNotAllowed.USER_IS_BEGINNER
+                        return True, "User allowed to validate"
+
+                user = UserService.get_user_by_id(user_id)
+
+                if not user.is_user_validator(user_id):
+                    return False, ValidatingNotAllowed.USER_NOT_VALIDATOR
```
Comment (willemarcel) [inline server/services/project_service.py]: do we need that part yet?

**Pull request merged**